### DISCUSSION TITLE
SDK for Java

### DISCUSSION THREAD

**Discussion by LucaPaterlini** [directus/directus#11147] [Feature Requests]
## Feature Request

Java SDK to interact with Directus

### Hello

Hello,
Where can I find a version of directus for java?

Some day you don't have the chance to choose the language you are working with :)

Thanks in advance for all the replies

**Comment by joselcvarela** [comment]
Hello,
We don't have any SDK for Java as of right now.
I have converted your original post to request this feature.

In the meantime you can use REST API for that. For example, crating an item (didn't test)
```java
URL url = new URL("https://directus.myserver.com/items/articles");
URLConnection con = url.openConnection();
HttpURLConnection http = (HttpURLConnection)con;
http.setRequestMethod("POST");
http.setDoOutput(true);
byte[] out = "{\"title\":\"Article 01\"}" .getBytes(StandardCharsets.UTF_8);
int length = out.length;
http.setFixedLengthStreamingMode(length);
http.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
http.connect();
try(OutputStream os = http.getOutputStream()) {
    os.write(out);
}
```

`https://directus.myserver.com` should be the address of your Directus instance
`articles` should be the collection used for creating a new item
`{"title": "Article 01"}` should be the body content of the item

**Reply by LucaPaterlini** [reply]
Thanks for your reply, what instead about the server side is there any chance to have it in have with directus or is there any open source project you are aware that do address the same problem in a similar manner?

**Reply by joselcvarela** [reply]
Are you asking for an alternative to Directus in Java?
Sorry, but this is not the right place to ask. Try reddit.com, stackoverflow.com or dev.to for example.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
